monkey/infection_monkey/network/network_scanner.py

```diff
@@ -68,49 +97,39 @@ def get_victim_machines(self, max_find=5, stop_callback=None):
         :param stop_callback: A callback to check at any point if we should stop scanning
         :return: yields a sequence of VictimHost instances
         """
-
-        TCPscan = TcpScanner()
-        Pinger = PingScanner()
+        pool = Pool()
         victims_count = 0
+        for victim_chunk in generate_victims(self._ranges, ITERATION_BLOCK_SIZE):
+            LOG.debug("Scanning for potential victims in chunk %r", victim_chunk)
 
-        for net_range in self._ranges:
-            LOG.debug("Scanning for potential victims in the network %r", net_range)
-            for ip_addr in net_range:
-                if hasattr(net_range, 'domain_name'):
-                    victim = VictimHost(ip_addr, net_range.domain_name)
-                else:
-                    victim = VictimHost(ip_addr)
-                if stop_callback and stop_callback():
-                    LOG.debug("Got stop signal")
-                    break
-
-                # skip self IP address
-                if victim.ip_addr in self._ip_addresses:
-                    continue
-
-                # skip IPs marked as blocked
-                if victim.ip_addr in WormConfiguration.blocked_ips:
-                    LOG.info("Skipping %s due to blacklist" % victim)
-                    continue
+            # skip self IP addresses
```

Comment:
1. WDYT about moving all the filtering to the generator? 
2. I think it might be inefficient to do multiple list comprehensions. Perhaps it will be better to create `is_self_ip_address`, `is_blocked_ip_address` methods and use them a `filter` call?